apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/McsClaimResponseTransformerV2.java

```diff
@@ -96,11 +99,22 @@ private static ClaimResponse transformClaim(PreAdjMcsClaim claimGroup) {
     claim.setContained(List.of(getContainedPatient(claimGroup)));
     claim.setExtension(getExtension(claimGroup));
     claim.setIdentifier(getIdentifier(claimGroup));
-    claim.setStatus(
-        CANCELLED_STATUS_CODES.contains(claimGroup.getIdrStatusCode().toLowerCase())
-            ? ClaimResponse.ClaimResponseStatus.CANCELLED
-            : ClaimResponse.ClaimResponseStatus.ACTIVE);
-    claim.setOutcome(OUTCOME_MAP.get(claimGroup.getIdrStatusCode()));
+
+    // Some status codes for MCS can be null.
+    // These will be interpreted as status=active, outcome=queued.
+    if (claimGroup.getIdrStatusCode() == null) {
+      claim.setStatus(ClaimResponse.ClaimResponseStatus.ACTIVE);
+      claim.setOutcome(ClaimResponse.RemittanceOutcome.QUEUED);
+    } else {
+      claim.setStatus(
+          STATUS_MAP.getOrDefault(
+              claimGroup.getIdrStatusCode().toLowerCase(),
+              ClaimResponse.ClaimResponseStatus.ACTIVE));
+      claim.setOutcome(
+          OUTCOME_MAP.getOrDefault(
+              claimGroup.getIdrStatusCode().toLowerCase(), ClaimResponse.RemittanceOutcome.QUEUED));
+    }
+
```

Comment:
Another thought for implementing this -- replace the two static final maps with static final Function<String, ClaimResponseStatus> and static final Function<String, RemittanceOutcome> and use those here. It keeps the calculation of status and outcome encapsulated in one place and gets rid of low level logic here in the transform method.

Reply:
It would still require a map (or a switch, etc.) to do the logic.  I can move them to a separate method though if that would help.